src/test/java/com/example/springboot/service/impl/FileReaderServiceImplTest.java

```diff
@@ -0,0 +1,35 @@
+package com.example.springboot.service.impl;
+
+import com.example.springboot.service.FileReaderService;
+import org.junit.jupiter.api.Test;
+import org.junit.Assert;
+
+import java.util.ArrayList;
+import java.util.List;
+
+class FileReaderServiceImplTest {
```

Comment:
let's add test cases for reading from the file that does not exist
and from the file that does not contain header

Reply:
Done